src/platform/Linux/ConfigurationManagerImpl.cpp

```diff
@@ -365,5 +365,25 @@ CHIP_ERROR ConfigurationManagerImpl::GetLocationCapability(uint8_t & location)
     return err;
 }
 
+CHIP_ERROR ConfigurationManagerImpl::GetCountryCode(char * buf, size_t bufSize, size_t & codeLen)
+{
+    CHIP_ERROR err = CHIP_NO_ERROR;
+
+    VerifyOrReturnError(bufSize > 2, CHIP_ERROR_BUFFER_TOO_SMALL);
+
+    if (!PosixConfig::ConfigValueExists(PosixConfig::kConfigKey_CountryCode))
```

Comment:
This test seems backwards?  If we don't have a value, why are we trying to read it?

Reply:
How the app know if the product has set the country code? the country code should be default to XX in case it is not set based on the Define XX as the special value for region agnostic use in Location attribute CHIP-Specifications/connectedhomeip-spec#4758, no?

Reply:
If the product has not set the  country code, if (!PosixConfig::ConfigValueExists(PosixConfig::kConfigKey_CountryCode)) should return true. We should return XX instead of empty string in this case, right?

Reply:
My point is that we seem to be doing `ReadConfigValueStr` if `ConfigValueExists` returns _false_.  And using "XX" when `ConfigValueExists` returns _true_.  That seems backwards, but maybe I am just totally misunderstanding something here?

Reply:
You are right, I mis-interpreted what you said